### PR TITLE
mediaconnect: fix flow and source arn format and update tests to use …

### PR DESCRIPTION
**Pull request by nickphbrown**
…valid name format

### CODE REVIEW

**Comment by nickphbrown** [comment]
I can't find any easy to read evidence of flow/source ARN format. However, it can be worked out from the example at the bottom of this page https://docs.aws.amazon.com/mediaconnect/latest/ug/flows-create-standard-source.html

where the relevant details in the example are:
- flow arn: "arn:aws:mediaconnect:us-east-1:111122223333:flow:1-23aBC45dEF67hiJ8-12AbC34DE5fG:AwardsShow"
implying the format for flow arn is "arn:aws:mediaconnect:{region}:{account_id}:flow:{id}:{flow_name}"

- source arn: "arn:aws:mediaconnect:us-east-1:111122223333:source:3-4aBC56dEF78hiJ90-4de5fG6Hi78Jk:AwardsShowSource"
implying the format for source arn is "arn:aws:mediaconnect:{region}:{account_id}:source:{id}:{source_name}"

Weirdly the id values in the example differ for the flow and the source ARNs, however from experience I've only ever seen the ids for the flow and its sources be the same. That observation may not always be true but I haven't seen an exception

File: tests/test_mediaconnect/test_mediaconnect.py [hunk 1]
```diff
@@ -209,13 +223,15 @@ def test_add_flow_vpc_interfaces_fails():
         client.add_flow_vpc_interfaces(FlowArn=flow_arn, VpcInterfaces=[])
     err = err.value.response["Error"]
     err["Code"].should.equal("NotFoundException")
-    err["Message"].should.equal("flow with arn=unknown-flow not found")
+    err["Message"].should.equal(
```
Comment (bblommers) [inline tests/test_mediaconnect/test_mediaconnect.py:226]: This looks like a dodgy merge - this was changed as part of our flake8-upgrade, but this PR reverses it again

File: tests/test_mediaconnect/test_mediaconnect.py [hunk 2]
```diff
@@ -254,13 +270,15 @@ def test_remove_flow_vpc_interface_fails():
         )
     err = err.value.response["Error"]
     err["Code"].should.equal("NotFoundException")
-    err["Message"].should.equal("flow with arn=unknown-flow not found")
+    err["Message"].should.equal(
```
Comment (bblommers) [inline tests/test_mediaconnect/test_mediaconnect.py:273]: Same

File: tests/test_mediaconnect/test_mediaconnect.py [hunk 3]
```diff
@@ -289,7 +307,9 @@ def test_add_flow_outputs_fails():
         client.add_flow_outputs(FlowArn=flow_arn, Outputs=[])
     err = err.value.response["Error"]
     err["Code"].should.equal("NotFoundException")
-    err["Message"].should.equal("flow with arn=unknown-flow not found")
+    err["Message"].should.equal(
```
Comment (bblommers) [inline tests/test_mediaconnect/test_mediaconnect.py:310]: Same

File: tests/test_mediaconnect/test_mediaconnect.py [hunk 4]
```diff
@@ -301,13 +321,15 @@ def test_remove_flow_output_fails():
         client.remove_flow_output(FlowArn=flow_arn, OutputArn=output_arn)
     err = err.value.response["Error"]
     err["Code"].should.equal("NotFoundException")
-    err["Message"].should.equal("flow with arn=unknown-flow not found")
+    err["Message"].should.equal(
```
Comment (bblommers) [inline tests/test_mediaconnect/test_mediaconnect.py:324]: Same

**Review by bblommers — CHANGES_REQUESTED** [review]
Yesterday we upgraded flake8 to 4.x - looks like this PR tries to reverse some of the changes, which causes the linter to fail.
https://github.com/spulec/moto/runs/6200952774?check_suite_focus=true

The change itself looks fine - we'll just need to the build to pass

File: tests/test_mediaconnect/test_mediaconnect.py (hunk 1 above)
Comment (nickphbrown) [reply tests/test_mediaconnect/test_mediaconnect.py]: woops, sorry, got myself a bit confused trying to get the testing/linting working locally. Thanks for pointing out the issue

**Review by bblommers — APPROVED** [review]
LGTM! Thanks for the fix @nickphbrown

**Pull request merged**